src/site/antora/modules/ROOT/pages/manual/async.adoc

```diff
@@ -14,525 +14,375 @@
--DAsyncLogger.WaitStrategy=busyspin (to use Async Loggers. The BusySpin
-wait strategy reduces some jitter.)
-* *classic mode:* -Dlog4j2.enable.direct.encoders=false +
-*garbage-free mode:* -Dlog4j2.enable.direct.encoders=true
-* -XX:CompileCommand=dontinline,org.apache.logging.log4j.core.async.perftest.NoOpIdleStrategy::idle
-* -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps
--XX:+PrintTenuringDistribution -XX:+PrintGCApplicationConcurrentTime
--XX:+PrintGCApplicationStoppedTime (to eyeball GC and savepoint pauses)
-
-The graph below compares the response time latency of the
-ArrayBlockingQueue-based asynchronous appenders in Logback 1.1.7, Log4j
-1.2.17 to the various options for asynchronous logging that Log4j 2.6
-offers. Under a workload of 128,000 messages per second, using 16
-threads (each logging at a rate of 8,000 messages per second), we see
-that Logback 1.1.7, Log4j 1.2.17 experience latency spikes that are
-orders of magnitude larger than Log4j 2.
-
-image:ResponseTimeAsyncLogging16Threads_8kEach.png[When 16
-threads generate a total workload of 128,000 msg/sec, Logback 1.1.7 and
-Log4j 1.2.17 experiences latency spikes that are orders of magnitude
-larger than Log4j 2]
-
-The graph below zooms in on the Log4j 2 results for the same test. We
-see that the worst-case response time is highest for the
-ArrayBlockingQueue-based Async Appender.
-xref:manual/garbagefree.adoc[Garbage-free] async loggers have the best response
-time behavior.
-
-image:ResponseTimeAsyncLogging16Threads_8kEachLog4j2Only-labeled.png[image]
+In order to enable it for a certain logger, set its
+xref:manual/configuration.adoc#logger-attributes-includeLocation[includeLocation] attribute to `true`.
+====
+
+[id=exception-handler]
+== Exception handler
+
+In order to handle exceptions that occur on the asynchronous thread, you can configure a custom
+https://lmax-exchange.github.io/disruptor/javadoc/com.lmax.disruptor/com/lmax/disruptor/ExceptionHandler.html[ExceptionHandler<T>].
+
+The exact type of handler depends on the configuration mode:
+
+Full asynchronous::
++
+If all the loggers are asynchronous you need to:
++
+* implement an
+link:../javadoc/log4j-core/org/apache/logging/log4j/core/async/RingBufferLogEvent.html[ExceptionHandler<? super RingBufferLogEvent>]
+* set its fully qualified class name as value of the
+xref:manual/systemproperties.adoc#log4j2.asyncLoggerExceptionHandler[log4j2.asyncLoggerExceptionHandler]
+configuration property.
+
+Mixed synchronous/asynchronous::
++
+If you use a mix of sync and async loggers you need to:
++
+* implement a
+link:../javadoc/log4j-core/org/apache/logging/log4j/core/async/AsyncLoggerConfigDisruptor.Log4jEventWrapper.html[ExceptionHandler<? super AsyncLoggerConfigDisruptor.Log4jEventWrapper>]
+* set its fully qualified class name as value of the
+xref:manual/systemproperties.adoc#log4j2.asyncLoggerConfigExceptionHandler[log4j2.asyncLoggerConfigExceptionHandler]
+configuration property.
 
 [#UnderTheHood]
```

Comment:
I'd just remove this section. We already do Disruptor marketing at the top and several other places where users need to plug in Disruptor components.

Reply:
Fixed in 13aea96008fe4d6757842f8a090b39f69c827384